Source/Core/Common/Logging/LogManager.cpp

```diff
@@ -204,11 +205,30 @@ void LogManager::Log(LogLevel level, LogType type, const char* file, int line, c
   LogWithFullPath(level, type, file + m_path_cutoff_point, line, message);
 }
 
+std::string LogManager::GetTimestamp()
+{
+  // NOTE: the Qt LogWidget hardcodes the expected length of the timestamp portion of the log line,
+  // so ensure they stay in sync
+
+  // We want milliseconds *and not hours*, so can't directly use STL formatters
+  auto ms =
+      std::chrono::time_point_cast<std::chrono::milliseconds>(std::chrono::system_clock::now())
+          .time_since_epoch();
+  auto secs = std::chrono::duration_cast<std::chrono::seconds>(ms);
+  ms -= std::chrono::duration_cast<std::chrono::milliseconds>(secs);
+  auto mins = std::chrono::duration_cast<std::chrono::minutes>(secs);
+  secs -= std::chrono::duration_cast<std::chrono::seconds>(mins);
+  auto hour = std::chrono::duration_cast<std::chrono::hours>(mins);
+  mins -= std::chrono::duration_cast<std::chrono::minutes>(hour);
+
+  return fmt::format("{:02}:{:02}:{:03}", mins.count(), secs.count(), ms.count());
```

Comment:
I think this would also work:

```
const auto now = std::chrono::system_clock::now();
const auto now_s = std::chrono::floor<std::chrono::seconds>(now);
const auto now_ms = std::chrono::floor<std::chrono::milliseconds>(now);
fmt::print("{:%M:%S}:{:03}", now_s, (now_ms - now_s).count());
````

Reply:
that is better, thanks!